### ISSUE TITLE
Misleading tarsnap error on negative account balance

### ISSUE THREAD

**Issue by ralish** [Tarsnap/tarsnap#370]
Assume the Tarsnap account associated with a given key has a negative balance. When attempting to perform operations against the account the following error is seen:
```
tarsnap: Network protocol violation by server
tarsnap: Error fscking archives
tarsnap: Error exit delayed from previous errors.
```

While a negative balance of course means Tarsnap should not be expected to work, the above error is extremely misleading. The [documentation on the error](https://www.tarsnap.com/network.html) also further reinforces this. In our case we missed a notification that our account needed credit added so weren't aware of the problem, and the Tarsnap error message sent us down entirely the wrong troubleshooting path.

We're using the latest Tarsnap package (1.0.39-2) from the APT repository on Ubuntu 18.04 x64. The command being run was an `--fsck` to rebuild the cache after migrating a service to a new server.

**Comment by jungle-boogie** [comment]
Hi,

See issue https://github.com/Tarsnap/tarsnap/issues/328

This was testing the exit code, but you'll notice the output message indicated what exactly to check. Must be an edge case that hasn't been checked for account balance before.

**Comment by ralish** [comment]
@jungle-boogie: Yeah, that sounds plausible. The output above is the entire total output but the exit code was non-zero (as it should be).

**Comment by cperciva** [comment]
Wow, this is odd and definitely something we should fix.  @gperciva can you reproduce this in your test environment?

**Comment by gperciva** [comment]
Confirmed:

```
$ ./tarsnap --keyfile local.keys --cachedir cache/ --fsck
tarsnap: Network protocol violation by server
tarsnap: Error fscking archives
tarsnap: Error exit delayed from previous errors.
```

Server log shows `tarsnap-server: Disallowing transaction start due to ENOMONEY`, so the problem is definitely in the client.  Fix incoming.

**Comment by gperciva** [comment]
This was fixed in #371.
